pkgs/applications/science/misc/root/default.nix

```diff
@@ -261,6 +261,14 @@ stdenv.mkDerivation rec {
     ]}"
   '';
 
+  # To use the debug information on the fly (without installation)
+  # add the outPath of root.debug into NIX_DEBUG_INFO_DIRS (in PATH-like format)
+  # and make sure that gdb from Nixpkgs can be found in PATH.
+  #
+  # Building with debug info and separating them is not supported on all systems
+  # e.g. Darwin fails to support it (#203380)
+  separateDebugInfo = stdenv.hostPlatform.isLinux;
```

Comment:
the conditional is not necessary, as separateDebugInfo = true; is silently ignored on darwin.

Reply:
It could hint users whether the `.debug` output is available or not.

It's quite surprising to see evaluation error about the missing `.debug` output when `.separateDebugInfo` is true.

Reply:
well if we start doing that everywhere in nixpkgs the day separateDebugInfo starts being available for darwin (which does not seem unlikely, thanks to this thread I discovered https://github.com/NixOS/nixpkgs/pull/188347 ) we will have to modify a lot of packages to benefit from it.